File: backend/blog/src/main/java/com/wofiporia/blog/model/Blog.java
package com.wofiporia.blog.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Lob
    @Column(columnDefinition = "TEXT")
    private String content;

    private LocalDateTime uploadDate;
    private LocalDateTime updateDate;

    // getter/setter
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public LocalDateTime getUploadDate() { return uploadDate; }
    public void setUploadDate(LocalDateTime uploadDate) { this.uploadDate = uploadDate; }
    public LocalDateTime getUpdateDate() { return updateDate; }
    public void setUpdateDate(LocalDateTime updateDate) { this.updateDate = updateDate; }
} 

File: backendNew/internal/model/blog.go
package model

import "time"

type Blog struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	Title      string    `json:"title"`
	Content    string    `gorm:"type:longtext" json:"content"`
	UploadDate time.Time `gorm:"column:upload_date" json:"uploadDate"`
	UpdateDate time.Time `gorm:"column:update_date" json:"updateDate"`
}

// TableName aligns with existing Java schema (table name: blog)
func (Blog) TableName() string { return "blog" }
